subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/AbstractCrossVersionGradleProfilerPerformanceTest.groovy

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance
+
+import org.gradle.integtests.fixtures.executer.IntegrationTestBuildContext
+import org.gradle.integtests.fixtures.executer.UnderDevelopmentGradleDistribution
+import org.gradle.integtests.fixtures.versions.ReleasedVersionDistributions
+import org.gradle.performance.categories.PerformanceRegressionTest
+import org.gradle.performance.fixture.GradleProfilerBuildExperimentRunner
+import org.gradle.performance.fixture.GradleProfilerCrossVersionPerformanceTestRunner
+import org.gradle.performance.fixture.PerformanceTestDirectoryProvider
+import org.gradle.performance.fixture.PerformanceTestIdProvider
+import org.gradle.performance.results.CrossVersionPerformanceResults
+import org.gradle.performance.results.CrossVersionResultsStore
+import org.gradle.performance.results.DataReporter
+import org.gradle.performance.results.SlackReporter
+import org.gradle.profiler.BenchmarkResultCollector
+import org.gradle.profiler.report.CsvGenerator
+import org.gradle.profiler.report.HtmlGenerator
+import org.gradle.test.fixtures.file.CleanupTestDirectory
+import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
+import org.junit.Rule
+import org.junit.experimental.categories.Category
+import spock.lang.Specification
+
+@Category(PerformanceRegressionTest)
+@CleanupTestDirectory
+class AbstractCrossVersionGradleProfilerPerformanceTest extends Specification {
```

Comment:
There should be some Javadoc on this and its sibling type to explain the difference (and the intention to use this one in future).